storage/merge.go

```diff
@@ -256,6 +264,7 @@ func (q *mergeGenericQuerier) LabelNames(ctx context.Context, hints *LabelHints,
 		labelNames = append(labelNames, name)
 	}
 	slices.Sort(labelNames)
+	labelNames = truncateToLimit(labelNames, hints)
```

Comment:
Why not truncate before adding all the strings to the slice?

Reply:
The merged results needs to be sorted before truncation.

Reply:
That's a restatement of the thing I asked about.
My question is: why?

Reply:
I think we could use the same merge sort algorithm used for LabelValues().